dashboard/app/controllers/pd/session_attendance_controller.rb

```diff
@@ -0,0 +1,74 @@
+class Pd::SessionAttendanceController < ApplicationController
+  load_and_authorize_resource :session, class: 'Pd::Session', id_param: :session_code, find_by: :code
+  rescue_from ActiveRecord::RecordNotFound do
+    render :not_found, status: :not_found
+  end
```

Comment:
This is rendering a specific (friendly) not_found page specific to this controller, from [this view](https://github.com/code-dot-org/code-dot-org/pull/15170/files#diff-292fcca9c117109bda4fbdbde95b84ab). By default, we get the generic [404 page](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/public/404.html).